xbmc/utils/SeekHandler.cpp

```diff
@@ -182,6 +182,33 @@ void CSeekHandler::Seek(bool forward, float amount, float duration /* = 0 */, bo
   m_timer.StartZero();
 }
 
+void CSeekHandler::SeekSeconds(int seconds)
+{
+  if (seconds == 0)
+    return;
+
+  m_requireSeek = true;
+  g_infoManager.SetSeeking(true);
+  m_seekStep = 0;
+  m_analogSeek = false;
+  m_seekDelay = 0;
+
+  if (g_infoManager.GetTotalPlayTime())
+    m_percentPlayTime = (float)g_infoManager.GetPlayTime() / g_infoManager.GetTotalPlayTime() * 0.1f;
+  else
+    m_percentPlayTime = 0.0f;
+
+  float percentPerSecond = 0.0f;
+  if (g_infoManager.GetTotalPlayTime())
+    percentPerSecond = 100.0f / (float)g_infoManager.GetTotalPlayTime();
+
+  m_percent = m_percentPlayTime + percentPerSecond * seconds;
```

Comment:
why is m_percentPlayTime a member variable and not a local? it's used in two methods but always reset at the start so no real reason to store state for it.